kore/src/Kore/Reachability/Prove.hs

```diff
@@ -324,19 +348,37 @@ proveClaim
             do
                 (proofDepth, proofState) <- acts
                 let maybeUnproven = extractUnproven proofState
-                for_ maybeUnproven $ \unproven -> do
+                for_ maybeUnproven $ \unproven -> lift $ do
                     infoUnprovenDepth proofDepth
-                    throwStuckClaim unproven
+                    stuckClaims <- State.get
+                    let updatedStuck =
+                            MultiAnd.singleton (StuckClaim unproven) <> stuckClaims
+                    when
+                        (MultiAnd.size updatedStuck >= maxCounterexamples)
+                        $ do
+                            Monad.Except.throwError updatedStuck
+                    State.put updatedStuck
```

Comment:
I think this part can be factored out and used in both this function and `throwStuckClaims`.